src/core/primitives.rs

```diff
@@ -128,11 +128,12 @@ fn j_hash(
     encapsulations: &[Encapsulation],
 ) -> Result<([u8; TAG_LENGTH], Secret<SHARED_SECRET_LENGTH>), Error> {
     let mut hasher = Shake::v256();
-
     hasher.update(seed);
-    for (c_i, seed_encapsulation) in c.iter().zip(encapsulations) {
+    for c_i in c.iter() {
```

Comment:
Also this bug has been corrected, that could allow for malleable encapsulations.